components/screens/DeployToken.tsx

```diff
@@ -0,0 +1,406 @@
+              <i
+                className="hn hn-info-circle text-[24px]"
+                data-tooltip-id="info-tooltip"
+                data-tooltip-content="A portion of the total tokens that are created and assigned to you (the deployer) right away."
+              ></i>
+              <Tooltip
+                id="info-tooltip"
+                style={{
+                  backgroundColor: "#374151",
+                  fontSize: "12px",
+                  fontWeight: 600,
+                  padding: "2px 8px",
+                }}
+              />
+              <label className="text-base text-[#E5E7EB]">
+                Preallocation{" "}
+                <span className="text-daintree-400">(Optional)</span>
+              </label>
+            </div>
+            <div className="flex flex-col gap-1">
+              <input
+                className={twMerge(
+                  "w-full rounded-lg border border-gray-700 bg-gray-800 px-4 py-3 text-white placeholder-gray-400 focus:outline-none",
+                  errors.preAllocation
+                    ? "border-red-700 focus:border-red-600 focus:ring-red-800"
+                    : "focus:border-gray-600 focus:ring-2 focus:ring-gray-500",
+                )}
+                {...register("preAllocation", {
+                  onChange: (event) => {
+                    event.target.value = event.target.value.replace(
+                      /[^0-9.]/g,
+                      "",
+                    );
+                  },
+                  valueAsNumber: true,
+                  max: {
+                    value: formattedMaxSupply,
+                    message: `Oh, the default mint amount should be at most ${formattedMaxSupply}`,
+                  },
+                })}
+                placeholder="0"
+              />
+              {errors.preAllocation && (
+                <span className="text-sm text-red-500">
+                  {errors.preAllocation.message}
+                </span>
+              )}
+            </div>
+          </div>
+
+          {/* Decimal places*/}
+          <div className="flex flex-col gap-3">
+            <div className="flex items-center gap-2">
+              <i
+                className="hn hn-info-circle text-[24px]"
+                data-tooltip-id="info-tooltip"
+                data-tooltip-content="The number of decimal points your token can be divided into."
+              ></i>
+              <Tooltip
+                id="info-tooltip"
+                style={{
+                  backgroundColor: "#374151",
+                  fontSize: "12px",
+                  fontWeight: 600,
+                  padding: "2px 8px",
+                }}
+              />
+              <label className="text-base text-[#E5E7EB]">
+                Decimal <span className="text-daintree-400">(Optional)</span>
+              </label>
+            </div>
+            <div className="flex flex-col gap-1">
+              <input
```

Comment:
According to the design, we should disable to edit decimal here.

Reply:
I am not sure I understand :/

Reply:
https://www.figma.com/design/0p1fiysKpbwjRsPp6cGCGX/Kastle?node-id=2766-162340

The design shows this field as disabled, and I remember we decided it should remain unchanged.

Reply:
I don't think it's disabled in the design you've linked since the field above has the same look.

As for the maximum, I have tested it for the maximum supply. You can go as high as you want if it doesn't explore the transaction payload maximum size XD.
I think it will be the same for decimal places but I should ask on discord.